lib/yext/api/utils/middleware/default_parameters.rb

```diff
@@ -7,51 +7,93 @@ module Middleware
         # Faraday Middleware for adding Configuration default values to API calls as needed.
         #
         # Most options can be set via scopes and the soped values will be kept if they are used.
+        # i.e. the existing query string values will be kept if they exist.
+        #
+        # Because Spyke puts `where` options into the body OR the query depending on the type of
+        # request being made, we have to move the Yext query options out of the body into the query
+        # if there is a body.
         #
         # The following configuration options are defaulted:
         #   validation_level: Added to query parameters if not already there and is not nil
         #   api_key:          Added to query parameters if not already there
         #   api_version:      Added to qeury parameters if not already there
         #   account_id:       Inserted into the path if the path includes the Account default URI and no
         #                     account_id was specified.
+        #
+        # The following values will be moved out of the body into the query string if they are found:
+        #   account_id      * Not used if in body.  If set it will already be in the path.
+        #   v
+        #   api_key
+        #   validation
+        #   yext_username
+        #   yext_user_id
         class DefaultParameters < Faraday::Response::Middleware
           attr_reader :configuration
 
           def call(request_env)
             @configuration = Yext::Api.configuration
             url            = request_env[:url]
             params         = Rack::Utils.parse_nested_query(url.query).with_indifferent_access
+            body_json      = parse_body(request_env)
 
-            add_username_headers(request_env, configuration, params)
-            add_default_query_params(configuration, params)
+            body_json.delete(:account_id)
+            add_username_headers(request_env, configuration, params, body_json)
+            add_default_query_params(configuration, params, body_json)
             save_query_params(url, params)
+            save_body(request_env, body_json)
 
             @app.call(request_env)
           end
 
           private
 
-          def add_default_query_params(configuration, params)
-            params.reverse_merge!(api_key: configuration.api_key,
-                                  v:       configuration.api_version || "20161012")
+          def parse_body(request_env)
+            if request_env.body.present?
+              JSON.parse(request_env.body, symbolize_names: true)
+            else
+              {}
+            end
+          end
+
+          def add_default_query_params(configuration, params, body_json)
+            params.reverse_merge!(api_key: body_json.delete(:api_key) || configuration.api_key,
+                                  v:       body_json.delete(:v) || configuration.api_version || "20161012")
```

Comment:
Should configuration return the default if nil? Seems odd to have that handled here.

Reply:
I thought about this too.

I had a reason for not doing it at one point, but I either can't remember what it was, or what it was doesn't seem reasonable any more.

Let me revisit this.

Reply:
I remember why I didn't do this.

I wanted to support `||=` for configuration.

```
Yext::Api::configuration.api_key ||= value
```

I can't do that if it returns a default value.
It may not be a great reason, but that is it.